otter/auth.py

```diff
@@ -61,6 +61,21 @@
 from otter.util.retry import repeating_interval, retry, retry_times
 
 
+class _DoNothingLogger(BoundLog):
```

Comment:
I think we already have something like this - `otter.test.utils`:  https://github.com/rackerlabs/otter/blob/master/otter/test/utils.py#L299

Would that work for these purposes?  It does call mock, so you can assert stuff about logging if you wish, but it also will eat all logging messages and you do not have to check them.  :)